src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -606,7 +606,11 @@ def set_operator(
         ],
         operator: str,
     ) -> "SelectStatement":
-        if isinstance(self.from_, SetStatement) and not self.has_clause:
+        if (
+            isinstance(self.from_, SetStatement)
+            and not self.has_clause
+            and not self.projection
```

Comment:
Is `self.has_clause` needed?  What is a case when `self.projection` is not None but `self.has_clause` is None?

Reply:
`has_clause` is `order by`, `where`,  `limit`, etc. They should be included because they impact the result.
An example is `df.filter()` creates a clause but not projection.

Reply:
I see, for example `session.table(...).filter(...)` does not set `SelectStatement.projection`. This makes sense.